presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -2068,18 +2098,34 @@ public void run(ExtendedHiveMetastore metastore)
                     // Not deleting the table may leave garbage behind. The former is much more dangerous than the latter.
                     // Therefore, the table is not considered added.
                 }
+
                 if (!done) {
                     throw e;
                 }
             }
+            tableCreated = true;
+        }
+
+        private boolean hasTheSameSchema(Table newTable, Table existingTable)
```

Comment:
I think this method is either incorrect or incorrectly named. It seems to be checking if the existing table is a superset of the new table (but is allowed to have extra columns).

Reply:
I guess It's the other way around (newTable can be a superset of existingTable based on this impl.) and I think I broke this during refactoring. This should check for equality. I will update this before merging.